solr/core/src/java/org/apache/solr/servlet/SolrDispatchFilter.java

```diff
@@ -213,7 +216,14 @@ private void setupJvmMetrics(CoreContainer coresInit)  {
       metricManager.registerAll(registryName, new OperatingSystemMetricSet(), SolrMetricManager.ResolutionStrategy.IGNORE, "os");
       metricManager.registerAll(registryName, new GarbageCollectorMetricSet(), SolrMetricManager.ResolutionStrategy.IGNORE, "gc");
       metricManager.registerAll(registryName, new MemoryUsageGaugeSet(), SolrMetricManager.ResolutionStrategy.IGNORE, "memory");
-      metricManager.registerAll(registryName, new ThreadStatesGaugeSet(), SolrMetricManager.ResolutionStrategy.IGNORE, "threads"); // todo should we use CachedThreadStatesGaugeSet instead?
+      if (System.getProperties().contains("solr.metrics.threads.caching.interval")) {
+        int threadsCachingInterval = Integer.valueOf(System.getProperty("solr.metrics.threads.caching.interval"));
+        log.info("Threads metrics will be cached for " + threadsCachingInterval + " seconds");
+        metricManager.registerAll(registryName, new CachedThreadStatesGaugeSet(threadsCachingInterval, TimeUnit.SECONDS), SolrMetricManager.ResolutionStrategy.IGNORE, "threads");
```

Comment:
is it possible to get only thread_count and runnable_thread?

Reply:
> is it possible to get only thread_count and runnable_thread?

I've updated the PR to disable the deadlock detection with a no-op impl, that should reduce the computation now. When this is disabled, threads.deadlocks.* metrics will be zero/empty.